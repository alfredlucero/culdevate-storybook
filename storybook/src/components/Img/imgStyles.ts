import { css } from 'styled-components';

const imgStyle = css`
  & img,
  & div {
    margin-right: 15px;
    width: 50px;
    height: 50px;
    border: solid 2px #eee;
  }

  & div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #eee;
  }
`;

export { imgStyle };
