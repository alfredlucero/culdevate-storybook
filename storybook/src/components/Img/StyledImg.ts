import styled from 'styled-components';

import { imgStyle } from './imgStyles';

export interface StyledImgProps {
  round?: boolean
}

const StyledImg = styled.div`
  ${imgStyle}

  & img,
  & div {
    border-radius: ${(props:StyledImgProps) => props.round ? '50%' : '5px'};
  }
`;

export { StyledImg };
