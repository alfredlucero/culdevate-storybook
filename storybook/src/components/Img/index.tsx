/**
 *
 * Img.ts
 *
 * Renders an image/default icon; given src, it will attempt to render
 * an img tag with alt text; otherwise, it will render a default icon type inside
 * (see types/icons.ts for all approved FontAwesome types)
 * There is also an option of choosing round 
 */

import * as React from 'react';
import styled from 'styled-components';

import { StyledImg } from './StyledImg';
import Icon from '../Icon';
import { IconType } from '../../types/icons';

export interface ImgProps {
  src: string;
  alt: string;
  defaultIcon: IconType;
  round?: boolean;
}

const Img: React.SFC<ImgProps> = ({
  src,
  alt,
  defaultIcon,
  round,
}) => {
  const imgChildren = src 
    ? <img src={src} alt={alt} />
    : <div><Icon type={defaultIcon} /></div>;

  return (
    <StyledImg round>
      {imgChildren}
    </StyledImg>
  )
};

export default Img;
