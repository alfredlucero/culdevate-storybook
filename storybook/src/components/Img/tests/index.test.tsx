import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Img from '../index';

describe('<Img />', () => {
  it('should render icon type correctly', () => {
    const tree = renderer.create(<Img src="" alt="" defaultIcon="users" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

