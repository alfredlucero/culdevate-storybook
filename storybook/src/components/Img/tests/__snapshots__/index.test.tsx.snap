// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should render icon type correctly 1`] = `
.c0 img,
.c0 div {
  margin-right: 15px;
  width: 50px;
  height: 50px;
  border: solid 2px #eee;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #eee;
}

.c0 img,
.c0 div {
  border-radius: 50%;
}

<div
  className="c0"
>
  <div>
    <i
      className="fa fa-users"
    />
  </div>
</div>
`;
